### PR TITLE
Validation webhook for Snow Datacenter Config

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*

*Description of changes:*
Validation webhook for Snow Datacenter Config

*Testing (if applicable):*
- Unit test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: config/manifest/eksa-components.yaml [hunk 1]
```diff
@@ -3199,6 +3199,20 @@ spec:
           status:
             description: CloudStackDatacenterConfigStatus defines the observed state
               of CloudStackDatacenterConfig
+            properties:
```
Comment (mitalipaygude) [inline config/manifest/eksa-components.yaml:3202]: This part was generated when I ran `make release-manifests`, it was probably something Cloudstack should've added but missed.

File: config/manifest/eksa-components.yaml [hunk 2]
```diff
@@ -3372,6 +3386,15 @@ spec:
           status:
             description: CloudStackMachineConfigStatus defines the observed state
               of CloudStackMachineConfig
+            properties:
```
Comment (mitalipaygude) [inline config/manifest/eksa-components.yaml:3389]: Same comment as above.

**Review by g-gaston — APPROVED** [review]

File: pkg/api/v1alpha1/snowdatacenterconfig_webhook.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var snowdatacenterconfiglog = logf.Log.WithName("snowdatacenterconfig-resource")
```
Comment (chrisdoherty4) [inline pkg/api/v1alpha1/snowdatacenterconfig_webhook.go:25]: Avoid package level state preferring to inject the logger instead.

**Pull request merged**